test/node/graphql-api/batched-queries.apollo.test.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @vitest-environment node
+ * Example of mocking batched GraphQL queries via Apollo.
+ * @see https://github.com/mswjs/msw/issues/510
+ * @see https://www.apollographql.com/docs/router/executing-operations/query-batching
+ */
+import { http, graphql, HttpResponse, getResponse } from 'msw'
+import { setupServer } from 'msw/node'
+
+const graphqlHandlers = [
+  graphql.query('GetUser', () => {
+    return HttpResponse.json({
+      data: {
+        user: { id: 1 },
+      },
+    })
+  }),
+  graphql.query('GetProduct', () => {
+    return HttpResponse.json({
+      data: {
+        product: { name: 'Hoover 2000' },
+      },
+    })
+  }),
+]
+
+export const handlers = [
+  // Let's create a request handler for batched GraphQL requests
+  // made using Apollo.
+  http.post('http://localhost/graphql', async ({ request, requestId }) => {
+    const data = await request.json()
+
+    // Ignore GraphQL queries that are not batched queries.
+    if (!Array.isArray(data)) {
+      return
+    }
+
+    const responses = await Promise.all(
+      // Resolve each individual query against the existing
+      // array of GraphQL request handlers.
+      data.map(async (operation) => {
+        const scopedRequest = new Request(request, {
+          body: JSON.stringify(operation),
+        })
+
+        const result = await getResponse({
```

Comment:
This is a much cleaner public API than `handleRequest`, which is relying heavily on the surrounding context (needs `emitter`, `options`).